lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -102,6 +104,14 @@ def upsert_records(messages)
         import_associations(valid_upserts) unless @association_list.blank?
       end
 
+      def remove_associations(messages)
```

Comment:
So thinking about this a bit more... this is probably unnecessary in every case because this is really just to cover the case where a previously existing association is no longer in the message. It probably makes sense to move this functionality to `import_associations`with the following basic structure:

*   Get all association keys from the payload
*   Do the import
*   Delete from the association where the key is NOT in the list of keys we got

Otherwise we'll be deleting and re-creating data every time.